CovertInfiltration/Src/CovertInfiltration/Classes/X2TemplateModifier.uc

```diff
@@ -0,0 +1,126 @@
+//---------------------------------------------------------------------------------------
+//  AUTHOR:  statusNone
+//  PURPOSE: Modifys Templates in order to make infinite items single build instead
+//---------------------------------------------------------------------------------------
+//  WOTCStrategyOverhaul Team
+//---------------------------------------------------------------------------------------
+class X2TemplateModifier extends Object config (SingleBuildItems);
+
+struct SingleBuildItem
+{
+	var name SchematicName;
+	var int TradingPostValue;
+	
+	var array<StrategyCost> Costs;
+};
+
+var config array<SingleBuildItem> SingleBuildItems;
+
+function ModifyTemplates()
+{
+	local array<X2DataTemplate> DifficultyVariants;
+	local X2ItemTemplateManager TemplateManager;
+	local X2SchematicTemplate Schematic;
+	local X2ItemTemplate Template;
+	local SingleBuildItem Item;
+	local int index, ValidIndex;
+
+	TemplateManager = class'X2ItemTemplateManager'.static.GetItemTemplateManager();
+	
+	foreach default.SingleBuildItems(Item)
+	{
+		Schematic = X2SchematicTemplate(TemplateManager.FindItemTemplate(Item.SchematicName));
+		Template = TemplateManager.FindItemTemplate(Item.SchematicName);
+		if (Schematic != none)
+		{
+			// cycle through and modify each difficulty variant for this template
+			TemplateManager.FindDataTemplateAllDifficulties(Item.SchematicName, DifficultyVariants);
+			for (index = 0; index < DifficultyVariants.Length - 1; index++)
+			{
+				Schematic = X2SchematicTemplate(DifficultyVariants[index]);
+				
+				ValidIndex = index;
+				// if no cost value was set for this difficulty variant use next highest value
```

Comment:
> use next highest value

Seems like it's the reverse?

Reply:
its not, it iterates downward checking each structure as it goes using the next highest value it can find